optuna/integration/wandb.py

```diff
@@ -0,0 +1,144 @@
+from typing import Optional
+from typing import Union
+
+import optuna
+from optuna._experimental import experimental
+from optuna._imports import try_import
+
+
+with try_import() as _imports:
+    import wandb
+
+
+@experimental("2.9.0")
+class WeightsAndBiasesCallback(object):
+    """Callback to track Optuna trials with Weights & Biases.
+
+    This callback enables tracking of Optuna study in
+    Weights & Biases. The study is tracked as a single experiment
+    run, where all suggested hyperparameters and optimized metric
+    are logged and plotted as a function of optimizer steps.
+
+    .. note::
+        User needs to be logged in to Weights & Biases before
```

Comment:
How about passing `mode` argument for `wandb.init`? Then we can use the W&B callback offline.

Reply:
Sure, thanks for the suggestion!